src/model.cpp

```diff
@@ -221,6 +222,78 @@ size_t Model::unitsCount() const
     return mPimpl->mUnits.size();
 }
 
+void linkComponentVariableUnits(const ComponentPtr &component)
+{
+    for (size_t index = 0; index < component->variableCount(); ++index) {
+        auto v = component->variable(index);
+        auto u = v->units();
+        if (u != nullptr) {
+            auto model = owningModel(u);
+            if (model == nullptr && !isStandardUnitName(u->name())) {
+                model = owningModel(component);
+                auto modelUnits = model->units(u->name());
+                if (modelUnits == nullptr) {
+                    model->addUnits(u);
+                    modelUnits = u;
+                }
+                v->setUnits(modelUnits);
+            }
+        }
+    }
+}
+
+void traversComponentTreeLinkingUnits(const ComponentPtr &component)
+{
+    linkComponentVariableUnits(component);
+    for (size_t index = 0; index < component->componentCount(); ++index) {
+        auto c = component->component(index);
+        traversComponentTreeLinkingUnits(c);
+    }
+}
+
+void Model::linkUnits()
+{
+    for (size_t index = 0; index < componentCount(); ++index) {
+        auto c = component(index);
+        traversComponentTreeLinkingUnits(c);
+    }
+}
+
+bool areComponentVariableUnitsUnlinked(const ComponentPtr &component)
+{
+    bool unlinked = false;
+    for (size_t index = 0; index < component->variableCount() && !unlinked; ++index) {
+        auto v = component->variable(index);
+        auto u = v->units();
+        if (u != nullptr) {
+            auto model = owningModel(u);
+            unlinked = model == nullptr && !isStandardUnitName(u->name());
+        }
+    }
+
+    return unlinked;
+}
+
+bool traveseComponentTreeForUnlinkedUnits(const ComponentPtr &component)
```

Comment:
ok, now I'm starting to think these are typos that were not picked up in code-completion tools :)

Reply:
Fixed too.